client.html

```diff
@@ -0,0 +1,55 @@
+<!DOCTYPE html>
+<html>
+
+<head>
+	<meta charset="utf-8">
+	<link href="diff-stylesheet.css" rel="stylesheet">
+	<script type="text/javascript">
+		'use strict';
+
+		function onNewContent() {
+			var httpRequest = ASMDW.httpRequest;
+			if (httpRequest.readyState === XMLHttpRequest.DONE) {
+				if (httpRequest.status === 200) {
+					document.getElementById('content').innerHTML = httpRequest.responseText;
+					if (ASMDW.requestForever) {
+						setTimeout(requestContent, 100);
+					}
+				}
+				else {
+					if (window.confirm('Failed to communicate with localhost:8000, close tab?')) {
+						window.close();
+					}
+				}
+			}
+		}
+
+		function requestContent() {
+			var httpRequest = new XMLHttpRequest();
+			ASMDW.httpRequest = httpRequest;
+			httpRequest.onreadystatechange = onNewContent;
+			httpRequest.open('GET', 'http://localhost:8000');
+			httpRequest.send();
+		}
+
+		function onBodyLoaded() {
+			requestContent();
+		}
+
+		// asm-differ web
+		window.ASMDW = {};
+		var url = new URL(window.location);
+		ASMDW.requestForever = url.searchParams.get('once') === null;
```

Comment:
not a fan of needing the `once` param, should ideally be part of the response instead. Potentially as a header if it's hard to fit in the body. Alternatively we can let the JS long-poll and just never get an update, which seems harmless and may make the code simpler, and would work better with the "red status marker" idea mentioned above

Reply:
Yeah about this and the last comment (about red status marker) I just need to improve on how client-server transactions are made, it's currently not displaying status such as "Building" or "Unchanged" in-browser either